README.md

```diff
@@ -0,0 +1,269 @@
+## Generic Bucket Provisioning
+
+Kubernetes natively supports dynamic provisioning for many types of file and
+block storage, but lacks support for object bucket provisioning. 
+This repo is a temporary placeholder for bucket provisioning CRDs and related
+generated client code. The longer term goal is to move this repo to a Kubernetes
+repo such as sig-storage or an external storage repo.
+
+### Design Overview
+S3 compatible object store buckets are stable and consistent among most, if not
+all, object store solutions. Therefore, the time has come where we can
+support a bucket provisioning API similar to that used for Persistent Volumes.
+We propose two new Custom Resources to abstract an object store bucket and
```

Comment:
It's important to also mention that this design does not (for now at least) include its own operator.  This is a key difference between our API and PV/PVCs. Binding between PV/PVCs is handled by 
a k8s core controller.  We don't provided that functionality.

Reply:
Agree. A generic design would be better if the client could somehow create the CM and Secret, based on parameters provided by the object store specific operators, so that we have _forced_ consistency in the CM and Secret, Otherwise, there's a higher chance that, if the Storage Class's provisioner is changed to a different object store, the app pod is affected.